### PR TITLE
Simplify checking format strings

### PR DESCRIPTION
**Pull request by abonie**
Seems like checking for end line flex and whether string is interpolated is not necessary for splitting format string into fragments, although I am not entirely sure about either of those.

### CODE REVIEW

**Review by psfinaki — COMMENTED** [review]
Are there already tests for this?

File: src/Compiler/Checking/CheckFormatStrings.fs [hunk 1]
```diff
@@ -56,60 +56,45 @@ let escapeDotnetFormatString str =
...
-                // Take of the ending quote or '{', always length 1
-                let fragLength = mLength - 1 - min endIndexFlex 1
-                (1, sourceText.GetSubTextString(startIndex + 1, fragLength), m)
-            else
-                // Take of the ending quote or '{', always length 1
-                let fragLength = mLength - 1 - min endIndexFlex 1
-                (1, sourceText.GetSubTextString(startIndex + 1, fragLength), m)
-        else (1, fmt, m) ]
+
+    let (|PrefixedBy|_|) (prefix: string) (str: string) =
```
Comment (vzarytovskii) [inline src/Compiler/Checking/CheckFormatStrings.fs:68]: Probably better to use struct active pattern here, to avoid allocations

**Comment by abonie** [comment]
> Are there already tests for this?

Good question. There are various tests related to this, but I am not 100% confident that their coverage is good enough. I'll check it out, although atm struggling with building fsharp locally

**Review by KevinRansom — APPROVED** [review]
Looks good.

**Review by T-Gro — APPROVED** [review]

**Pull request merged**